gremlin-go/driver/performance-tests/java/PerformanceTests.java

```diff
@@ -0,0 +1,200 @@
+    final GraphTraversalSource g = traversal().withRemote(connection);
+    final int REPEATS = 50;
+
+    public static GraphTraversal<?, ?> getTraversal(final String name, final GraphTraversalSource g) {
+        final Map<String, GraphTraversal<?, ?>> traversals = new HashMap<>();
+        traversals.put("getSongVertices", g.V().hasLabel("song"));
+        traversals.put("getNumVertices", g.V().count());
+        return traversals.get(name);
+
+    }
+
+    public static String[] getArgs(final int repeats) {
+        final String[] args = new String[repeats];
+        for (int i = 0; i < repeats; i++) {
+            args[i] = Integer.toString(i);
+        }
+        return args;
+    }
+
+    public static GraphTraversal<?, ?> getProjectTraversal(final GraphTraversalSource g, final int repeats, final String[] args) {
+        GraphTraversal<?, ?> traversal = g.V().project(args[0], Arrays.copyOfRange(args, 1, repeats));
+        for (int i = 0; i < repeats; i++) {
+            traversal = traversal.by(__.valueMap(true));
+        }
+        return traversal;
+    }
+
+    @Test
+    public void sample() {
+        System.out.println("~~~~~~~ RUNNING ONE ITEM PERFORMANCE TEST ~~~~~~~");
+        executeGetOneItemPerformanceTest();
+        System.out.println("~~~~~~~ RUNNING LIST PERFORMANCE TEST ~~~~~~~");
+        executeGetItemListPerformanceTest();
+        System.out.println("~~~~~~~ RUNNING TRANSFER PERFORMANCE TEST ~~~~~~~");
+        executeTransferPerformanceTest();
+        System.out.println("~~~~~~~ PERFORMANCE TESTS COMPLETE ~~~~~~~\"");
+    }
+
+    public timingData executeGetOneItemPerformanceTest() {
+        final List<Duration> durations = new ArrayList<>();
+        for (int i = 0; i < SAMPLE_SIZE; i++) {
+            try {
+                final String[] args = getArgs(REPEATS);
+                final Instant starts = Instant.now();
+                final Object temp = getProjectTraversal(g, REPEATS, args).next();
+                final Instant ends = Instant.now();
+                durations.add(Duration.between(starts, ends));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        timingData data = getTimingDataFromDurationList(durations);
+        System.out.println("Retrieve One Item (.next) performance test results:");
+        System.out.println(data.toStringMillis("One Item"));
+        return data;
+    }
+
+    public timingData executeGetItemListPerformanceTest() {
+        final List<Duration> durations = new ArrayList<>();
+        for (int i = 0; i < SAMPLE_SIZE; i++) {
+            try {
+                final String[] args = getArgs(REPEATS);
+                final Instant starts = Instant.now();
+                final List<?> list = getProjectTraversal(g, REPEATS, args).toList();
+                for (Object item : list) {
+                    final Object temp = item;
```

Comment:
What this variable is for? Never used?

Reply:
code is updated to not store results in variable.